wlp-managed-8.5/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -197,57 +215,81 @@ public void run() {
 
                if (itse == null)
                   throw new LifecycleException("Process terminated prematurely; ev = " + ev);
-               
+
                if (vmid == null)
                   // Find WebSphere Liberty Profile VMs by looking for ws-launch.jar and the name of the server
                   vmid = findVirtualMachineIdByName(containerConfiguration.getServerName());
-               
+
                if (wlpvm == null && vmid != null)
                   wlpvm = VirtualMachine.attach(vmid);
-               
+
                if (serviceURL == null && wlpvm != null)
                   serviceURL = getVMLocalConnectorAddress(wlpvm);
             }
-            
+
             // If serviceURL is still null, we were unable to start the virtual machine
             if (serviceURL == null)
                throw new LifecycleException("Unable to retrieve connector address for localConnector of started VM");
-            
+
             log.finer("vmid: " + vmid);
          }
       } catch (Exception e) {
          throw new LifecycleException("Could not start container", e);
       }
-      
+
       try {
          JMXServiceURL url = new JMXServiceURL(serviceURL);
          jmxConnector = JMXConnectorFactory.connect(url);
          mbsc = jmxConnector.getMBeanServerConnection();
       } catch (IOException e) {
          throw new LifecycleException("Connecting to the JMX MBean Server failed", e);
       }
-      
+
       if (log.isLoggable(Level.FINER)) {
          log.exiting(className, "start");
       }
    }
 
+	private List<String> parseJvmArgs(String javaVmArguments) {
+		List<String> parsedJavaVmArgumetns = new ArrayList<String>();
+		String[] splitJavaVmArguments = javaVmArguments.split(javaVmArgumentsDelimiter);
+		if (splitJavaVmArguments.length > 1) {
+			for (String javaVmArgument : splitJavaVmArguments) {
+				if (javaVmArgument.trim().length() > 0) {
+					// remove precessing spaces
+					if(javaVmArgument.startsWith(javaVmArgumentsIndicator)) {
+						// vm argument without spaces
+						parsedJavaVmArgumetns.add(javaVmArgument);
+					} else {
```

Comment:
Well, from my point of view, it's necessary because the property value might contain spaces and then the simplified splitting approached won't work correctly (e.g.: -Dblabal=some spaced property would become [-Dblabal=some] [spaced] [property]) and the JVM won't start. I think this is likely to happen, especially if properties refer to a path in the file-system